runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt

```diff
@@ -24,15 +24,24 @@ import kotlinx.cinterop.*
 /**
  * Class representing worker.
  */
-@Frozen
-public class Worker private constructor(val id: Int) {
+@Suppress("NON_PUBLIC_PRIMARY_CONSTRUCTOR_OF_INLINE_CLASS")
+public inline class Worker @PublishedApi internal constructor(val id: Int) {
     companion object {
         /**
          * Start new scheduling primitive, such as thread, to accept new tasks via `execute` interface.
          * Typically new worker may be needed for computations offload to another core, for IO it may be
          * better to use non-blocking IO combined with more lightweight coroutines.
          */
-        public fun start(): Worker = Worker(startInternal())
+        public fun start(errorReporting: Boolean = true): Worker = Worker(startInternal(errorReporting))
+
+        /**
+         * Return current worker, if known, null otherwise, for example, in the main thread or in platform
+         * thread without an associated worker.
+         */
+        public val current: Worker? get() {
```

Comment:
The documentation is not clear enough to make a conclusion that `Worker.current != null` only in a thread that was created with `Worker.start()`.

Reply:
Somewhat reworded.